konduit-serving-core/src/main/java/ai/konduit/serving/launcher/command/KonduitRunCommand.java

```diff
@@ -0,0 +1,96 @@
+ */
+
+package ai.konduit.serving.launcher.command;
+
+import ai.konduit.serving.util.LogUtils;
+import io.vertx.core.cli.CLIException;
+import io.vertx.core.cli.annotations.*;
+import io.vertx.core.impl.launcher.CommandLineUtils;
+import io.vertx.core.impl.launcher.commands.RunCommand;
+import org.nd4j.shade.guava.base.Strings;
+
+import java.io.File;
+import java.util.Collections;
+
+import static ai.konduit.serving.launcher.KonduitServingLauncher.*;
+
+@Name(value = "run", priority = 1)
+@Summary("Runs a konduit server in the foreground.")
+@Description("Runs a konduit server in the foreground.")
+public class KonduitRunCommand extends RunCommand {
+
+    public static final String DEFAULT_SERVICE = "inference";
+
+    @Override
+    @Option(longName = "service", shortName = "s", argName = "type")
+    @DefaultValue(DEFAULT_SERVICE)
+    @Description("Service type that needs to be deployed. Defaults to \"inference\"")
+    public void setMainVerticle(String konduitServiceType) {
+        if(getServicesMap().containsKey(konduitServiceType)) {
+            super.setMainVerticle(KONDUIT_PREFIX + ":" + konduitServiceType);
+        } else {
+            throw new CLIException(String.format("Invalid service type %s. " +
+                    "Allowed values are: %s", konduitServiceType,
+                    Collections.singletonList(getServicesMap().keySet())));
+        }
+    }
+
+    /**
+     * The main verticle configuration, it can be a json file or a json string.
+     *
+     * @param configuration the configuration
+     */
+    @Override
+    @Option(shortName = "c", longName = "config", argName = "config", required = true)
+    @Description("Specifies a configuration that should be provided to the verticle. <config> should reference either a " +
+            "text file containing a valid JSON object which represents the configuration OR be a JSON string.")
+    public void setConfig(String configuration) {
+        File file = new File(configuration);
+        if(file.exists()) {
+            configuration = file.getAbsolutePath();
+        }
+
+        log.info("Processing configuration: {}", configuration);
+        super.setConfig(configuration);
+    }
+
+    @Override
+    public void run() {
+        String serverId = getServerId();
+        log.info("Starting konduit server with an id of '{}'", serverId);
+
+        if(!Strings.isNullOrEmpty(serverId)) {
+            LogUtils.setAppendersForRunCommand(serverId);
+        }
+
+        super.run();
+
+        log.info("Starting konduit server with an id of '{}'", serverId);
+    }
+
+    private String getServerId() {
+        String[] commandSplits = CommandLineUtils.getCommand().split(" ");
+        String lastSegment = commandSplits[commandSplits.length - 1];
+        if(lastSegment.contains("serving.id")) {
+            return lastSegment.replace("-Dserving.id=", "").trim();
+        } else {
+            return null;
+        }
```

Comment:
This makes sure that it's at the end of the command line arguments: https://github.com/KonduitAI/konduit-serving/blob/03567a83bce210f2457ae22beb493b876ed319d2/konduit-serving-core/src/main/java/ai/konduit/serving/launcher/command/ServeCommand.java#L297

Also, since the command parser for the `serve` command would ignore spaces while taking arguments so we won't have to worry about that here.